### PR TITLE
Fix regression 4388

### LINKED ISSUE #4388: master branch orders some resource names differently than v4.4.1
**Describe the bug**

At v4.4.1 and before, kustomize build produces output with resource names sorted like so:

1. testing
2. testing-one
3. testing-two

In the master branch ~~(I believe first appearing at 92197fd)~~, kustomize build produces output with resource names sorted like so:

1. testing-one
2. testing-two
3. testing

Cross reference https://github.com/kubernetes-sigs/kustomize/pull/4361#issuecomment-1012374075

**Files that can reproduce the issue**

kustomization.yaml

```yaml
resources:
  - resources.yaml
```

resources.yaml

```
apiVersion: v1
kind: ConfigMap
metadata:
  name: testing
data:
  key: value
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: testing-one
data:
  key: value
---
apiVersion: v1
kind: ConfigMap
metadata:
  name: testing-two
data:
  key: value
```

**Expected output, using v4.4.1 as an example**

```
$ build/kustomize-execs/release-v4.4.1/kustomize version
{Version:kustomize/v4.4.1 GitCommit:b2d65ddc98e09187a8e38adc27c30bab078c1dbf BuildDate:2021-11-11T23:36:27Z GoOs:linux GoArch:amd64}
$ build/kustomize-execs/release-v4.4.1/kustomize build testdata/kust-resource-sort
apiVersion: v1
data:
  key: value
kind: ConfigMap
metadata:
  name: testing
---
apiVersion: v1
data:
  key: value
kind: ConfigMap
metadata:
  name: testing-one
---
apiVersion: v1
data:
  key: value
kind: ConfigMap
metadata:
  name: testing-two
$
```

**Actual output**

```
$ cat build/kustomize-execs/branch-master/info.yaml
branch: master
sha1: 6e82b210a9e0651cc8392d2931c64348137b82ff
$ build/kustomize-execs/branch-master/kustomize build testdata/kust-resource-sort
apiVersion: v1
data:
  key: value
kind: ConfigMap
metadata:
  name: testing-one
---
apiVersion: v1
data:
  key: value
kind: ConfigMap
metadata:
  name: testing-two
---
apiVersion: v1
data:
  key: value
kind: ConfigMap
metadata:
  name: testing
$
```

**Kustomize version**

This change is not (yet) in a released version. I see the change in the master branch, ~~and I believe it first appears at 92197fd~~.

**Platform**

Linux

### PR DESCRIPTION
**Pull request by KnVerey**
Fixes #4388 

This should have been release-blocking, and we should re-release today with this patch.
/cc @natasha41575

### CODE REVIEW

File: api/krusty/legacy_order_test.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
+package krusty_test
+
+import (
+	"testing"
+
+	kusttest_test "sigs.k8s.io/kustomize/api/testutils/kusttest"
+)
+
+func TestIssue4388(t *testing.T) {
+	th := kusttest_test.MakeHarness(t)
```
Comment (KnVerey) [inline api/krusty/legacy_order_test.go:10]: The test harness doesn't seem to run with legacy sort order by default, so this is not a proper regression test. Working on it.

File: kyaml/resid/gvk_test.go [hunk 2]
```diff
@@ -47,8 +47,8 @@ var lessThanTests = []struct {
 		Gvk{Group: "a", Version: "c", Kind: "ClusterRole"}},
 	{Gvk{Group: "a", Version: "c", Kind: "Namespace"},
 		Gvk{Group: "a", Version: "b", Kind: "ClusterRole"}},
-	{Gvk{Group: "b", Version: "c", Kind: "Namespace"},
-		Gvk{Group: "a", Version: "d", Kind: "Namespace"}},
+	{Gvk{Group: "a", Version: "d", Kind: "Namespace"},
+		Gvk{Group: "b", Version: "c", Kind: "Namespace"}},
```
Comment (KnVerey) [inline kyaml/resid/gvk_test.go:51]: This is a test case revert from https://github.com/kubernetes-sigs/kustomize/pull/4344/files#diff-148cb3bc0ba93ee6684c49539296bb7045b250be09b3b03cc5073a434319f995L50

**Comment by natasha41575** [comment]
Sorry I accidentally hit close with comment instead of comment

**Pull request merged**